presto-main/src/main/java/com/facebook/presto/sql/planner/EqualityInference.java

```diff
@@ -328,8 +329,8 @@ public EqualityPartition(Iterable<Expression> scopeEqualities, Iterable<Expressi
 
     public static class Builder
     {
-        private final Map<Expression, Expression> map = new HashMap<>();
-        private final Multimap<Expression, Expression> reverseMap = HashMultimap.create();
+        private final DisjointSet<Expression> unionFindSet = new DisjointSet<>();
```

Comment:
`unionFindSet` is not very descriptive. Maybe call it "equivalences", "equalities" or something like that?